tests/drivers/misc/stm32_tsc/src/test_stm32_tsc.c

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c) 2024 Arif Balik <arifbalik@outlook.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Verify STM32 TSC peripheral is configured properly and can be started
+ *
+ * @details
```

Comment:
A note here about the pin configuration required for the SYNC pin activation would be welcomed.